types/extract-files/index.d.ts

```diff
@@ -1,20 +1,32 @@
-// Type definitions for extract-files 3.1
+// Type definitions for extract-files 8.1
 // Project: https://github.com/jaydenseric/extract-files#readme
 // Definitions by: Edward Sammut Alessi <https://github.com/Slessi>
+//                 Alex K <https://github.com/lynxtaa>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
+export type ExtractableFileMatcher = (value: any) => boolean;
+
+export const isExtractableFile: ExtractableFileMatcher;
+
+export function extractFiles<T>(
+    value: T,
+    path?: string,
+    isExtractableFile?: ExtractableFileMatcher,
+): {
+    clone: T;
+    files: Map<any, string[]>;
```

Comment:
@lynxtaa why `any` ?

Reply:
Fixed. Now I'm using generics to support custom file matchers (https://github.com/jaydenseric/extract-files/pull/11)